flake8_trio.py

```diff
@@ -1590,6 +1590,20 @@ def is_p_wait(arg: ast.expr) -> bool:
             return
 
 
+@error_class
+class Visitor24X(Visitor200):
+    error_codes = {
+        "TRIO240": ("Avoid using os.path, prefer using trio.Path objects"),
+    }
+
+    def visit_Call(self, node: ast.Call):
+        if not self.async_function:
+            return
+        func_name = ast.unparse(node.func)
+        if func_name.startswith("os.path."):
```

Comment:
Alas, I suspect that requiring rewrites for all code that calls `os.path.join()` (and other pure functions) is not going to be viable, so we'll need to read through https://docs.python.org/3/library/os.path.html and work out the list.

Reply:
Read through the source code, and came up with this list
```python
    safe = (
        "_get_sep",
        "_varprog",
        "_varprogb",
        "normcase",
        "isabs",
        "join",
        "split",
        "splitext",
        "splitdrive",
        "basename",
        "dirname",
        "expanduser",
        "commonpath",
        "expandvars",  # os.fsencode, os.environ, os.fsdecode
        "abspath",  # calls, os.getcwd, which I think python tracks internally?
        "commonprefix",
        "samestat",  # just does simple equality check
        # constants
        "curdir",
        "pardir",
        "extsep",
        "sep",
        "pathsep",
        "defpath",
        "altsep",
        "devnull",
        "supports_unicode_filenames",
    )
    os_funcs = (
        "_path_normpath",
        "normpath",
        "_joinrealpath",
        "islink",
        "lexists",
        "ismount",  # safe on windows, unsafe on posix
        "realpath",
        "exists",
        "isdir",
        "isfile",
        "getatime",
        "getctime",
        "getmtime",
        "getsize",
        "samefile",
        "sameopenfile",
        "relpath",
    )
```